packages/Takeover/src/components/OverlayPropsCollector.js

```diff
@@ -0,0 +1,5 @@
+export default function OverlayPropsCollector() {
```

Comment:
interesting component name. makes sense i guess, i'd imagine we'll see more components that serve this purpose so we'd want to use the same naming convention for the others in the future.

Reply:
Yeah, we had a naming discussion in a PR for the `<DatePicker>` a while back. I think it was @DanilAgafonov's PR...

Reply:
Can y'all remind me why we need this at all? So far only seeing the displayName rehashing.

Reply:
It's all for the API.  We do this:

https://github.com/acl-services/paprika/blob/126258c17692d4be62acd346d28979ac4ffc3b93/packages/Takeover/src/Takeover.js#L57

And this:

https://github.com/acl-services/paprika/blob/126258c17692d4be62acd346d28979ac4ffc3b93/packages/Takeover/src/Takeover.js#L68

Then we can pass props to the underlying `<Overlay>` by doing `<Takeover.Overlay someProp={someValue} />` instead of the undesirable `<Takeover someOverlayProp={someValue} />` which can get pretty ugly as it scales.  

Historically I think maybe the `<Listbox>` was the first to adopt the pattern – we wanted to pass a number of props to the underlying `<Popover>` and didn't want a `popoverProps` prop.

Reply:
Thanks. 👍 
FWIW I think this is unnecessary. The same should be possible by just having `Takeover.Overlay = Overlay`, and then this
https://github.com/acl-services/paprika/blob/126258c17692d4be62acd346d28979ac4ffc3b93/packages/Takeover/src/Takeover.js#L67-L89
would instead be
```jsx
React.cloneElement(overlayExtracted, { "data-pka-anchor": "takeover.overlay", ... }, state => (
  ...
))
```
Seems like a less roundabout way to achieve the same.

Reply:
I like "less roundabout ways", thanks for the suggestion. I will give it a try.

Reply:
Surfacing Overlay this way has some interesting semver implications though. I.e., now if Takeover bumps their Overlay version across a major, it becomes a major bump for Takeover itself, because the API of Overlay is a subset of Takeover's API.